src/trigger.rs

```diff
@@ -1106,16 +1112,22 @@ pub fn validate_target_ok_this(
         warn(ErrorKey::Scopes).msg(msg).loc(part).opt_loc_msg(opt_loc, msg2).push();
     }
     sc.close();
+    final_scopes
 }
 
 /// Just like [`validate_target_ok_this`], but warns if the target is a literal `this` because that
 /// is usually a mistake.
-pub fn validate_target(token: &Token, data: &Everything, sc: &mut ScopeContext, outscopes: Scopes) {
-    validate_target_ok_this(token, data, sc, outscopes);
+pub fn validate_target(
+    token: &Token,
+    data: &Everything,
+    sc: &mut ScopeContext,
+    outscopes: Scopes,
+) -> Scopes {
     if token.is("this") {
         let msg = "target `this` makes no sense here";
         warn(ErrorKey::UseOfThis).msg(msg).loc(token).push();
     }
+    validate_target_ok_this(token, data, sc, outscopes)
```

Comment:
I think we can skip this in the else branch

Reply:
I still want to check if `this` is the right scope :)